pymongo/ssl_support.py

```diff
@@ -19,11 +19,23 @@
 
 from pymongo.errors import ConfigurationError
 
+import warnings
+
```

Comment:
This import needs to go above `from typing import Optional` in order to pass our pre-commit checks: https://github.com/mongodb/mongo-python-driver/blob/master/CONTRIBUTING.md#running-linters

```
import warnings
from typing import Optional
```

Reply:
But of course. Could be more efficient to let CI run early to save manual review time. It seems contributors aren't permitted to use the checks themselves, which could help improving the coding style without manual intervention. I would suggest a bit more relaxed policy to allow contributors use CI after they have added something sane i.e. just rule out vandalism.

However, the mentioned pre-commit lint warning should now be solved.

Reply:
Hi @oh2fih, the GitHub options are limited, we're using a balanced approach:

<img width="717" alt="image" src="https://github.com/mongodb/mongo-python-driver/assets/2096628/c887d78a-c5d9-4e3d-b0fa-2f33ef2f0ec6">

Reply:
Thanks @blink1073 ; GitHub recognizes me as a first-time contributor since this is my first PR and not yet merged.